src/shared/test/java/com/couchbase/lite/internal/core/C4BaseTest.java

```diff
@@ -259,11 +259,11 @@ private long importJSONLines(InputStream is, String idPrefix, double timeout, bo
         boolean commit = false;
         db.beginTransaction();
         try {
-            BufferedReader br = new BufferedReader(new InputStreamReader(is));
+            BufferedReader br = new BufferedReader(new InputStreamReader(is, "UTF-8"));
             try {
                 String line;
                 while ((line = br.readLine()) != null) {
-                    FLSliceResult body = db.encodeJSON(line.getBytes());
+                    FLSliceResult body = db.encodeJSON(line.getBytes("UTF-8"));
```

Comment:
Jay, can you have a look at:
```
jstring litecore::jni::UTF8ToJstring(JNIEnv *env, char *s, size_t size)
```
... and 
```
std::string litecore::jni::JstringToUTF8(JNIEnv *env, jstring jstr) {
```
... to see if they are related, at all.  I don't like the idea that we might be doing the same thing in multiple ways.  Especially on different platforms, the implementations are sure to be different...

Reply:
do we need to create separate JNI method for tests, which is not giving much benefits.